src/Sulu/Bundle/TestBundle/Testing/TestUserProvider.php

```diff
@@ -160,4 +161,9 @@ public function supportsClass($class)
     {
         return $this->userProvider->supportsClass($class);
     }
+
+    public function reset(): void
+    {
+        $this->user = null;
+    }
```

Comment:
Symfony now resets the kernel between requests. As this means the entity manager unit of work is being resetd we also need to reset the user here that it is loaded again into Unit of work next time it is requested.